flink-yarn/src/main/java/org/apache/flink/yarn/AbstractYarnClusterDescriptor.java

```diff
@@ -483,12 +483,14 @@ private void validateClusterSpecification(ClusterSpecification clusterSpecificat
 		}
 
 		final int yarnMinAllocationMB = yarnConfiguration.getInt(YarnConfiguration.RM_SCHEDULER_MINIMUM_ALLOCATION_MB, 0);
+		final int yarnSchedulerMaxVcores = yarnConfiguration.getInt(YarnConfiguration.RM_SCHEDULER_MAXIMUM_ALLOCATION_VCORES, 0);
```

Comment:
We do not need to get the scheduler max vcores from yarn configuration. Use `maximumResourceCapability.getVirtualCores()` instead.

Reply:
Thats a very different value. maximumResourceCapability.getVirtualCores() Specifies how much total vcores are available. That doesn't mean that a container of that size can be allocated.

Yarn has a different setting for the maximum container vcore size it can allocate.

Reply:
In any case if we can't get this value from the yarn config we should default to int maxvalue so as to not block the deployment

Reply:
@gyfora I think you misunderstood my point. The `maximumResourceCapability` what i mean is `GetNewApplicationResponse.getMaximumResourceCapability()`. I have checked the yarn code and write a unit test to confirm. The `maximumResourceCapability.getVirtualCores()` returns the value of the configuration `yarn.scheduler.maximum-allocation-vcores` of yarn scheduler.

Get the max allocation vcores from yarn configuration in flink client in not very good. We should better get it from the yarn RM side.

Reply:
@wangyang0918 thanks for your feedback. Please correct me if I am wrong. AFAIK, maximumResourceCapability.getVirtualCores() seems to get the aggregated amount of cores from YARN node managers that are across the YARN cluster. But the value of yarn.scheduler.maximum-allocation-vcores in yarn-site.xml is still governing the maximum of vcores per container that flink can get from YARN. 

In other words, if maximum-allocation-vcores is capped at 4, even the total cores in yarn cluster is plenty. Flink can only get container with no more than 4 cores.